packages/projection/src/operators/certificates/withCertificates.ts

```diff
@@ -26,16 +26,18 @@ const blockCertificates = ({
     body
   }
 }: WithBlock) =>
-  body.flatMap(({ body: { certificates = [] } }, txIndex) =>
-    certificates.map((certificate, certIndex) => ({
-      certificate,
-      pointer: {
-        certIndex,
-        slot,
-        txIndex
-      }
-    }))
-  );
+  body
+    .filter((tx) => tx.isValid !== false)
```

Comment:
Great point, I agree - we probably don't need both properties for OnChainTx because they represent the same thing. Maybe let's keep only 1 of them, but
- it's optional for `Tx`
- it's required for `OnChainTx`

I personally like inputSource terminology better.

Reply:
On a second thought, 'isValid' terminology makes more sense when the use case is to bypass validation when submitting the transaction. Maybe let's keep it after all, but `Omit<Tx, 'isValid'>` for `OnChainTx`? :thinking: need more opinions here @rhyslbw @AngelCastilloB @mirceahasegan

Reply:
I'm not sure about what I should do here. I assume there's nothing to do. `isValid` is set by the user while `inputSource` is the status of the transaction.

Reply:
I would update `OnChainTx` type to Omit<Tx, 'isValid'>

Reply:
[ac373cd](https://github.com/input-output-hk/cardano-js-sdk/pull/607/commits/ac373cd29e9dc1ec9786a7a09c93b115b6fb7378)